sopel_modules/weather/wz.py

```diff
@@ -114,7 +114,13 @@ def get_current(self, city, state, weather):
         )
         if 'alerts' in weather:
             result += " | Alerts: "
-            result += ', '.join([x['title'] + ' ' + self.__short(x['uri']) for x in weather['alerts']])
+            uris = set()
+            alerts = []
+            for x in weather['alerts']:
+                if x['uri'] not in uris:
+                    alerts.append(x)
+                    uris.add(x['uri'])
```

Comment:
Couldn't this just be like `uris = set([x['uri'] for x in weather['alerts']])` ?

```
>>> weather = {}
KeyboardInterrupt
>>> weather['alerts'] = [ {'uri': 'foo'}, {'uri': 'bar'}, {'uri': 'bin'}, {'uri: 'foo'}, {'uri': 'foo'} ]
>>> set([x['uri'] for x in weather['alerts']])
{'foo', 'bar', 'bin'}
```

Reply:
The set is just there to filter the alerts, the set is not used after the filtering is done.

Reply:
You could do something like
seen = set()
alerts = [x for x in weather['alerts'] if x['uri'] not in seen and not seen.add(x['uri'])]

which i just found, but it's a lil odd to have a comprehension have side effects, to me.

Reply:
Your call